### PR TITLE
APPS-472 Wait on upload, dxCompiler

### PR DESCRIPTION
**Pull request by r-i-v-a**
PR to develop branch this time.

Needs to be merged after the corresponding PR in dxScala for https://jira.internal.dnanexus.com/browse/APPS-472 -- https://github.com/dnanexus/dxScala/pull/7.

Also need to publish the -SNAPSHOT version from dxScala/develop, since this one depends on it.

### CODE REVIEW

File: executorCommon/src/main/scala/dx/executor/FileUploader.scala [hunk 1]
```diff
@@ -27,14 +29,14 @@ trait FileUploader {
...
+  def upload(files: Set[Path], wait: Boolean = false): Map[Path, DxFile] = {
+    files.map(path => path -> dxApi.uploadFile(path, wait = wait)).toMap
   }
 
-  def upload(files: Map[Path, String]): Map[Path, DxFile] = {
+  def uploadWithDestination(files: Map[Path, String], wait: Boolean = false): Map[Path, DxFile] = {
     files.map {
       case (path, destination) =>
-        (path, dxApi.uploadFile(path, Some(destination)))
+        (path, dxApi.uploadFile(path, Some(destination), wait))
```
Comment (jdidion) [inline executorCommon/src/main/scala/dx/executor/FileUploader.scala:39]: Name the wait parameter (`wait = wait`)

File: executorCommon/src/main/scala/dx/executor/TaskExecutor.scala [hunk 2]
```diff
@@ -458,11 +459,11 @@ abstract class TaskExecutor(jobMeta: JobMeta,
     val delocalizedPathToUri: Map[Path, String] = {
       val dxFiles = if (jobMeta.useManifests) {
         // if using manifests, we need to upload the files directly to the project
-        fileUploader.upload(delocalizingValueToPath.values.map { path =>
+        fileUploader.uploadWithDestination(delocalizingValueToPath.values.map { path =>
           path -> s"${jobMeta.manifestFolder}/${path.getFileName.toString}"
-        }.toMap)
+        }.toMap, waitOnUpload)
```
Comment (jdidion) [inline executorCommon/src/main/scala/dx/executor/TaskExecutor.scala:464]: Same here

File: executorCwl/src/main/scala/dx/executor/cwl/CwlTaskExecutor.scala [hunk 3]
```diff
@@ -51,8 +53,9 @@ object CwlTaskExecutor {
 case class CwlTaskExecutor(tool: CommandLineTool,
                            jobMeta: JobMeta,
                            fileUploader: FileUploader,
-                           streamFiles: StreamFiles)
-    extends TaskExecutor(jobMeta, fileUploader, streamFiles) {
+                           streamFiles: StreamFiles,
+                           waitOnUpload: Boolean)
+    extends TaskExecutor(jobMeta, fileUploader, streamFiles, waitOnUpload) {
```
Comment (jdidion) [inline executorCwl/src/main/scala/dx/executor/cwl/CwlTaskExecutor.scala:58]: same here

**Review by jdidion — CHANGES_REQUESTED** [review]

File: executorCommon/src/main/scala/dx/executor/TaskExecutor.scala (hunk 2 above)
Comment (r-i-v-a) [reply executorCommon/src/main/scala/dx/executor/TaskExecutor.scala]: done

File: executorCwl/src/main/scala/dx/executor/cwl/CwlTaskExecutor.scala (hunk 3 above)
Comment (r-i-v-a) [reply executorCwl/src/main/scala/dx/executor/cwl/CwlTaskExecutor.scala]: done

File: test/upload_file/upload_file.wdl [hunk 4]
```diff
@@ -0,0 +1,20 @@
+version 1.1
```
Comment (r-i-v-a) [inline test/upload_file/upload_file.wdl:1]: @jdidion   Added minimal WDL file that uploads a file here. Still need to add integration test that calls dxCompiler with the flag.

Addressed other PR comments.

Comment (jdidion) [reply test/upload_file/upload_file.wdl:1]: Looks good. To add an integration test, go to scripts/run_tests.py and add the test name (without path or .wdl extension) to the appropriate test set.

**Review by jdidion — APPROVED** [review]

**Pull request merged**